metrics/internal/collectors/storageconsumer.go

```diff
@@ -0,0 +1,99 @@
+package collectors
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	"github.com/red-hat-storage/ocs-operator/metrics/internal/options"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+var _ prometheus.Collector = &StorageConsumerCollector{}
+
+type StorageConsumerCollector struct {
+	Informer                cache.SharedIndexInformer
+	StorageConsumerMetadata *prometheus.Desc
+	AllowedNamespace        string
+}
+
+type StorageConsumerLister interface {
+	List(selector labels.Selector) (ret []*ocsv1alpha1.StorageConsumer, err error)
+}
+
+type storageConsumerLister struct {
+	indexer cache.Indexer
+}
+
+func NewStorageConsumerLister(indexer cache.Indexer) StorageConsumerLister {
+	return &storageConsumerLister{indexer: indexer}
+}
+
+func (s *storageConsumerLister) List(selector labels.Selector) (ret []*ocsv1alpha1.StorageConsumer, err error) {
+	err = cache.ListAll(s.indexer, selector, func(m interface{}) {
+		ret = append(ret, m.(*ocsv1alpha1.StorageConsumer))
+	})
+	return ret, err
+}
```

Comment:
How about we move this alongside ocs-client code? Will keep the collector implementation cleaner and easier to read.

Reply:
Done